src/common/target-helper.ts

```diff
@@ -78,14 +78,12 @@ export class TargetHelper {
         return shadowDomSelectors;
     };
 
-    public static getSelectorFromTarget = (target: Target): string | undefined => {
-        if (target) {
-            return target
-                .map((targets: string | string[]) =>
-                    typeof targets === 'string' ? targets : targets.join(','),
-                )
-                .join(';');
-        }
+    public static getSelectorFromTarget = (target: Target): string => {
```

Comment:
Won't this still return `undefined` if `target` is `undefined`?

Reply:
Yes, intentionally; there are still a bunch of indirect usages in non-strict-null files right now (mostly via `assessment-builder`) and I haven't analyzed all of them yet, so I wanted to simultaneously:

* keep the original behavior intact to avoid breaking those cases accidentally (that's why there's still unit tests for this behavior as well)
* strengthen the typing so that cases that *are* strict-null-checked and can verify that they don't pass undefined targets will also understand that they won't get back undefined selectors given that they're using it right.

I'll add a comment to that effect, I agree it isn't obvious enough from just reading the code that this would be intentional

Reply:
(and I'll do the same for `getTargetFromSelector`, for the same reasons)